hypothesis-python/src/hypothesis/stateful.py

```diff
@@ -544,8 +568,21 @@ def available(self, data):
 
 
 class BundleConsumer(Bundle[Ex]):
-    def __init__(self, bundle: Bundle[Ex]) -> None:
-        super().__init__(bundle.name, consume=True)
+
+    def __init__(self, bundle: Bundle[Ex], **kwargs) -> None:
+        super().__init__(bundle.name, consume=True, **kwargs)
+
+    def filter(self, condition):
+        return type(self)(
+            self,
+            transformations=(*self._transformations, ("filter", condition)),
+        )
+
+    def map(self, pack):
+        return type(self)(
+            self,
+            transformations=(*self._transformations, ("map", pack)),
+        )
```

Comment:
This is now so similar to `Bundle` that I think we don't need a separate class anymore; instead of checking `isinstance(x, BundleConsumer)` we can just check `isinstance(x, Bundle) and x.consume`.